src/mlx_omni_server/chat/mlx/mlx_model.py

```
@@ -386,27 +416,131 @@ def stream_generate(
             )
             logger.debug(f"Encoded prompt:\n{prompt}")
 
+            # Thinking tokens handling
             completion = ""
+            accumulated_text = ""
+            in_thinking_mode = False
+            thinking_content = ""
+            had_thinking_tag = False
```

Ruff (0.11.9)

Local variable `had_thinking_tag` is assigned to but never used

Remove assignment to unused variable `had_thinking_tag`

(F841)